lib/floe/runner/container_runner.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative "container_runner/docker_mixin"
+require_relative "container_runner/docker"
+require_relative "container_runner/kubernetes"
+require_relative "container_runner/podman"
+
+module Floe
+  class Runner
+    class ContainerRunner
+      class << self
+        def cli_options(optimist)
+          optimist.banner("")
+          optimist.banner("Container runner options:")
+
+          optimist.opt :container_runner, "Type of runner for docker container images (docker, podman, or kubernetes)", :type => :string, :short => 'r'
+          optimist.opt :container_runner_options, "Options to pass to the container runner", :type => :strings, :short => 'o'
+
+          optimist.opt :docker,     "Use docker to run container images     (short for --container-runner=docker)",     :type => :boolean
+          optimist.opt :podman,     "Use podman to run container images     (short for --container-runner=podman)",     :type => :boolean
+          optimist.opt :kubernetes, "Use kubernetes to run container images (short for --container-runner=kubernetes)", :type => :boolean
+        end
+
+        def resolve_cli_options!(opts)
+          # shortcut support
+          opts[:container_runner] ||= "docker" if opts[:docker]
+          opts[:container_runner] ||= "podman" if opts[:podman]
+          opts[:container_runner] ||= "kubernetes" if opts[:kubernetes]
+
+          runner_options = opts[:container_runner_options].to_h { |opt| opt.split("=", 2) }
+
```

Comment:
accepting options and how the options are formatted is an optimist problem.

So I think that is part of the cli layer

Handling adding a container runner name parameter is very container runner. so it belongs as part of the container runner optimist extension. (how this extension enhances the ui)

Not sure the final answer. So that means this is good enough

Reply:
Yeah I agree - the "common" cli options parsing can be pushed down to a shared method.  I can most definitely do that in a follow up.